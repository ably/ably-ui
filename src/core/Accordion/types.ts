import { ReactNode } from "react";
import { IconName, IconSize } from "../Icon/types";
import { ColorThemeSet } from "../styles/colors/types";

/**
 * Represents the data structure for an Accordion component.
 */
export type AccordionData = {
  /**
   * The name of the accordion item.
   */
  name: string;

  /**
   * The optional icon name to be displayed alongside the accordion item.
   */
  icon?: IconName;

  /**
   * The content to be displayed when the accordion item is expanded.
   */
  content: ReactNode;

  /**
   * Optional click handler function that is called when the accordion item is clicked.
   * @param index - The index of the clicked accordion item.
   */
  onClick?: (index: number) => void;

  /**
   * Indicates whether the accordion item is interactive.
   * When false, the item cannot be expanded or collapsed by user interaction.
   * @default true
   */
  interactive?: boolean;
};

export type AccordionIcons = {
  closed: {
    name: IconName;
    css?: string;
  };
  open: {
    name: IconName;
    css?: string;
  };
};

export const accordionThemes = ["default", "transparent", "static"] as const;

export type AccordionTheme = (typeof accordionThemes)[number];

/**
 * Represents the theme colors for an accordion component.
 */
export type AccordionThemeColors = {
  /**
   * Background color class for the accordion.
   */
  bg: ColorThemeSet;

  /**
   * Background color when the accordion item is hovered.
   */
  hoverBg: ColorThemeSet;

  /**
   * Text color class for the accordion.
   */
  text: ColorThemeSet;

  /**
   * Color class for the toggle icon of the accordion.
   */
  toggleIconColor: ColorThemeSet;

  /**
   * Optional background color class for selectable accordion items.
   */
  selectableBg?: ColorThemeSet;

  /**
   * Optional text color class for selectable accordion items.
   */
  selectableText?: ColorThemeSet;

  /**
   * Optional border color for the accordion.
   */
  border?: string;
};

/**
 * Options for configuring the Accordion component.
 */
export type AccordionOptions = {
  /**
   * If true, only one accordion item can be open at a time.
   * @default false
   */
  autoClose?: boolean;

  /**
   * If true, accordion items can be selected.
   * @default false
   */
  selectable?: boolean;

  /**
   * If true, the accordion header will stick to the top when scrolling.
   * @default false
   */
  sticky?: boolean;

  /**
   * An array of indexes indicating which accordion items should be open by default.
   * @default []
   */
  defaultOpenIndexes?: number[];

  /**
   * If true, all accordion items will be fully open.
   * @default false
   */
  fullyOpen?: boolean;

  /**
   * Custom CSS class to apply to the accordion header.
   * @default ""
   */
  headerCSS?: string;

  /**
   * If true, borders between accordion items will be hidden.
   * @default false
   */
  hideBorders?: boolean;

  /**
   * Size of the row icon.
   * @default "32px"
   */
  rowIconSize?: IconSize;

  /**
   * Size of the accordion icon.
   * @default "16px"
   */
  iconSize?: IconSize;

  /**
   * Custom CSS classes to apply to the selected accordion header.
   * @default ""
   */
  selectedHeaderCSS?: string;

  /**
   * Custom CSS classes to apply to the accordion content.
   * @default ""
   */
  contentCSS?: string;
};
