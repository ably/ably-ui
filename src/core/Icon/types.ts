import { computedIcons } from "./computed-icons";

export const iconNames = {
  gui: computedIcons.gui,
  display: computedIcons.display,
  social: computedIcons.social,
  other: computedIcons.other,
  tech: computedIcons.tech,
  product: computedIcons.product,
};

export type IconName =
  | (typeof iconNames.gui)[number]
  | (typeof iconNames.display)[number]
  | (typeof iconNames.social)[number]
  | (typeof iconNames.other)[number]
  | (typeof iconNames.tech)[number]
  | (typeof iconNames.product)[number];
