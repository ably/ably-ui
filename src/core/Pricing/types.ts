import { ReactNode } from "react";
import { ColorClass, ColorThemeSet } from "../styles/colors/types";

type PricingDataHeader = {
  content: string;
  className?: string;
  color?: ColorClass | ColorThemeSet;
  tooltip?: string | ReactNode;
};

type PricingDataFeatureCta = {
  text: string;
  url: string;
  className?: string;
  disabled?: boolean;
  onClick?: () => void;
  iconColor?: ColorClass | ColorThemeSet;
  isPriority?: boolean;
};

export type PricingDataFeatureSection = {
  title: string;
  items: string[] | string[][];
  listItemColors?: {
    foreground: ColorClass | ColorThemeSet;
    background: ColorClass | ColorThemeSet;
  };
};

export type PricingDataFeatureBorder = {
  text: string;
  style: "border-solid" | "border-dashed" | "border-dotted";
  color: "neutral" | "blue" | "orange";
};

export type PricingDataFeature = {
  title: PricingDataHeader;
  description: PricingDataHeader;
  price: { amount: string; content?: ReactNode };
  cta?: PricingDataFeatureCta;
  sections: PricingDataFeatureSection[];
  border?: PricingDataFeatureBorder;
  subtext?: string;
  bottomCta?: PricingDataFeatureCta;
};
