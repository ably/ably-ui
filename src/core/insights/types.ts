export type InsightsConfig = {
  debug: boolean;
  mixpanelToken: string;
  mixpanelAutoCapture: boolean;
  mixpanelRecordSessionsPercent: number;
  posthogApiKey: string;
  posthogApiHost: string;
};

// Define the interface for our analytics service
export interface AnalyticsService {
  initInsights: (config: InsightsConfig) => void;
  enableDebugMode: () => void;
  disableDebugMode: () => void;
  identify: (identity: InsightsIdentity) => void;
  trackPageView: (options?: TrackPageViewOptions) => void;
  track: (event: string, properties?: Record<string, unknown>) => void;
  startSessionRecording: () => void;
  stopSessionRecording: () => void;
  setupObserver: () => () => void;
}

// Command type for our queue
export type Command = {
  methodName: keyof AnalyticsService;
  args: unknown[];
};

export type InsightsIdentity = {
  userId: string;
  accountId: string;
  organisationId?: string;
  email?: string;
  name?: string;
};

export type TrackPageViewOptions = {
  includeDataLayer?: boolean;
  excludeIds?: string[];
} & Record<string, unknown>;
