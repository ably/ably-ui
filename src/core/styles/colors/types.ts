export type ColorName =
  | (typeof neutralColors)[number]
  | (typeof orangeColors)[number]
  | (typeof secondaryColors)[number]
  | (typeof guiColors)[number]
  | (typeof aliasedColors)[number]
  | "ably-primary"
  | "ably-primary-inverse"
  | "ably-secondary"
  | "ably-secondary-inverse"
  | "ably-tertiary"
  | "ably-tertiary-inverse"
  | "ably-label"
  | "ably-label-inverse";

export const variants = ["", "dark:"] as const;

type ColorClassVariants = (typeof variants)[number];

export const prefixes = ["text", "bg", "from", "to", "border"] as const;

type ColorClassPrefixes = (typeof prefixes)[number];

export const colors = [
  "neutral",
  "orange",
  "blue",
  "yellow",
  "green",
  "violet",
  "pink",
] as const;

export type ColorClassColorGroups = (typeof colors)[number];

export type Theme = "light" | "dark";

export type ColorClass =
  `${ColorClassVariants}${ColorClassPrefixes}-${ColorName}`;

export type ColorThemeSet = `${string} dark:${string}`;

export const neutralColors = [
  "neutral-000",
  "neutral-100",
  "neutral-200",
  "neutral-300",
  "neutral-400",
  "neutral-500",
  "neutral-600",
  "neutral-700",
  "neutral-800",
  "neutral-900",
  "neutral-1000",
  "neutral-1100",
  "neutral-1200",
  "neutral-1300",
] as const;

export const orangeColors = [
  "orange-100",
  "orange-200",
  "orange-300",
  "orange-400",
  "orange-500",
  "orange-600",
  "orange-700",
  "orange-800",
  "orange-900",
  "orange-1000",
  "orange-1100",
] as const;

export const yellowColors = [
  "yellow-100",
  "yellow-200",
  "yellow-300",
  "yellow-400",
  "yellow-500",
  "yellow-600",
  "yellow-700",
  "yellow-800",
  "yellow-900",
] as const;

export const greenColors = [
  "green-100",
  "green-200",
  "green-300",
  "green-400",
  "green-500",
  "green-600",
  "green-700",
  "green-800",
  "green-900",
] as const;

export const blueColors = [
  "blue-100",
  "blue-200",
  "blue-300",
  "blue-400",
  "blue-500",
  "blue-600",
  "blue-700",
  "blue-800",
  "blue-900",
] as const;

export const violetColors = [
  "violet-100",
  "violet-200",
  "violet-300",
  "violet-400",
  "violet-500",
  "violet-600",
  "violet-700",
  "violet-800",
  "violet-900",
] as const;

export const pinkColors = [
  "pink-100",
  "pink-200",
  "pink-300",
  "pink-400",
  "pink-500",
  "pink-600",
  "pink-700",
  "pink-800",
  "pink-900",
] as const;

const secondaryColors = [
  ...yellowColors,
  ...greenColors,
  ...blueColors,
  ...violetColors,
  ...pinkColors,
] as const;

const guiColors = [
  "gui-blue-default-light",
  "gui-blue-hover-light",
  "gui-blue-active-light",
  "gui-blue-default-dark",
  "gui-blue-hover-dark",
  "gui-blue-active-dark",
  "gui-blue-focus",
  "gui-unavailable",
  "gui-success-green",
  "gui-error-red",
  "gui-focus",
  "gui-focus-outline",
  "gui-visited",
] as const;

export const aliasedColors = [
  "white",
  "extra-light-grey",
  "light-grey",
  "mid-grey",
  "dark-grey",
  "charcoal-grey",
  "cool-black",
  "active-orange",
  "bright-red",
  "red-orange",
  "electric-cyan",
  "zingy-green",
  "jazzy-pink",
  "gui-default",
  "gui-hover",
  "gui-active",
  "gui-error",
  "gui-success",
  "gui-default-dark",
  "gui-hover-dark",
  "gui-active-dark",
  "transparent",
] as const;

export const colorRoles = {
  neutral: neutralColors,
  orange: orangeColors,
  secondary: secondaryColors,
  gui: guiColors,
};

export const colorGroupLengths = {
  neutral: neutralColors.length,
  orange: orangeColors.length,
  blue: blueColors.length,
  yellow: yellowColors.length,
  green: greenColors.length,
  violet: violetColors.length,
  pink: pinkColors.length,
};
