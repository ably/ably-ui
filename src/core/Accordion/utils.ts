import { AccordionTheme, AccordionThemeColors } from "./types";

export const themeClasses: Record<AccordionTheme, AccordionThemeColors> = {
  default: {
    bg: "bg-neutral-200 dark:bg-neutral-1100",
    hoverBg: "hover:bg-neutral-300 dark:hover:bg-neutral-1100",
    text: "text-ably-primary",
    toggleIconColor: "text-neutral-1000 dark:text-orange-600",
    selectableBg: "bg-neutral-1200 dark:bg-neutral-300",
    selectableText: "text-ably-primary-inverse",
  },
  transparent: {
    bg: "bg-transparent dark:bg-transparent",
    hoverBg: "hover:bg-transparent dark:hover:bg-transparent",
    text: "text-neutral-1000 dark:text-neutral-000",
    toggleIconColor: "text-dark-grey dark:text-orange-600",
    border:
      "border-neutral-500 border-b last:border-none dark:border-neutral-1000",
  },
  static: {
    bg: "bg-neutral-200 dark:bg-neutral-1200",
    hoverBg: "hover:bg-neutral-200 dark:hover:bg-neutral-1200",
    text: "text-ably-primary",
    toggleIconColor: "text-neutral-200 dark:text-neutral-1200",
    selectableBg: "bg-neutral-1200 dark:bg-neutral-1200",
    selectableText: "text-ably-primary-inverse",
  },
};

export const isNonTransparentTheme = (theme: AccordionTheme) =>
  theme !== "transparent";

export const isStaticTheme = (theme: AccordionTheme) => theme === "static";
