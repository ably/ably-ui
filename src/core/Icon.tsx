import React, { CSSProperties, useCallback, useId, useMemo } from "react";

import {
  defaultIconSecondaryColor,
  getHeroicon,
  setUniqueIds,
  toPascalCase,
} from "./Icon/utils";
import * as IconComponents from "./Icon/components";
import { IconName, IconSize } from "./Icon/types";
import { ColorClass, ColorThemeSet } from "./styles/colors/types";
import { convertTailwindClassToVar } from "./styles/colors/utils";
import cn from "./utils/cn";

export type IconProps = {
  name: IconName;
  size?: IconSize;
  color?: ColorClass | ColorThemeSet;
  secondaryColor?: ColorClass | ColorThemeSet;
  additionalCSS?: string;
  variant?: "outline" | "solid" | "micro" | "mini";
};

// Helper function to get a heroicon component from the static map
const getHeroiconComponent = (
  iconName: string,
): React.ComponentType<React.SVGProps<SVGSVGElement>> | null => {
  // Parse the icon name to extract heroicon name and variant
  // Expected format: icon-gui-{heroicon-name}-{variant}
  const match = iconName.match(/^icon-gui-(.+)-(outline|solid|mini|micro)$/);
  if (!match) {
    return null;
  }

  const [, heroiconName, variant] = match;
  const component = getHeroicon(heroiconName, variant);
  return component;
};

const Icon = ({
  name,
  size = "0.75rem",
  color,
  secondaryColor,
  additionalCSS = "",
  variant,
  ...additionalAttributes
}: IconProps) => {
  const uniqueId = useId();

  const [lightSecondaryColor, darkSecondaryColor] = (
    secondaryColor ?? ""
  ).split(" dark:") as [ColorClass, ColorClass | undefined];

  const iconName =
    variant && !name.endsWith(`-${variant}`) ? `${name}-${variant}` : name;

  // Get the local icon component (memoized)
  const LocalIconComponent = useMemo(():
    | React.ComponentType<React.SVGProps<SVGSVGElement>>
    | undefined => {
    return IconComponents[
      toPascalCase(iconName) as keyof typeof IconComponents
    ];
  }, [iconName]);

  // Get the heroicon component only if no local component is found (memoized)
  const HeroiconComponent = useMemo(() => {
    return LocalIconComponent ? null : getHeroiconComponent(iconName);
  }, [LocalIconComponent, iconName]);

  const setUniqueIdsRef = useCallback(
    (el: SVGSVGElement | null) => {
      setUniqueIds(el, uniqueId);
    },
    [uniqueId],
  );

  const iconSvg = useCallback(
    (secondaryColor?: ColorClass, isDark?: boolean, isThemed?: boolean) => {
      if (LocalIconComponent) {
        let secondaryColorValue;
        if (secondaryColor) {
          secondaryColorValue = convertTailwindClassToVar(secondaryColor);
        } else if (defaultIconSecondaryColor(name)) {
          secondaryColorValue = convertTailwindClassToVar(
            defaultIconSecondaryColor(name),
          );
        }

        return (
          <LocalIconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
              {
                "hidden dark:block": secondaryColor && !isDark && isThemed,
              },
              {
                "dark:hidden": secondaryColor && isDark && isThemed,
              },
            )}
            style={
              {
                width: size,
                height: size,
                ...(secondaryColorValue && {
                  "--ui-icon-secondary-color": secondaryColorValue,
                }),
              } as CSSProperties
            }
            {...additionalAttributes}
            ref={setUniqueIdsRef}
          />
        );
      }

      if (HeroiconComponent) {
        return (
          <HeroiconComponent
            className={cn(
              { [`${color}`]: color },
              { [`${additionalCSS}`]: additionalCSS },
            )}
            style={{
              width: size,
              height: size,
            }}
            {...additionalAttributes}
          />
        );
      }
    },
    [
      LocalIconComponent,
      HeroiconComponent,
      iconName,
      size,
      color,
      additionalCSS,
      lightSecondaryColor,
      darkSecondaryColor,
      name,
    ],
  );

  if (lightSecondaryColor && darkSecondaryColor) {
    return (
      <>
        {iconSvg(lightSecondaryColor, false, !!darkSecondaryColor)}
        {iconSvg(darkSecondaryColor, true, true)}
      </>
    );
  } else if (lightSecondaryColor) {
    return iconSvg(lightSecondaryColor, false, !!darkSecondaryColor);
  } else {
    return iconSvg();
  }
};

export default Icon;
