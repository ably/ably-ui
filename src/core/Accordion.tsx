import React, { useState, ReactNode, JSX } from "react";
import Icon from "./Icon";

type AccordionData = {
  name: string;
  content: ReactNode;
};

type AccordionRowProps = {
  bottomBorder?: boolean;
  topBorder?: boolean;
  active: boolean;
  last: boolean;
  name: string;
  index: number;
  children: ReactNode;
  arrowIcon?: boolean;
  setActiveIndex: (index: number) => void;
};

export type AccordionProps = {
  data: AccordionData[];
  arrowIcon?: boolean;
  topBorder?: boolean;
  bottomBorder?: boolean;
  id?: string;
  autoClose?: boolean;
  className?: string;
};

const AccordionRow = ({
  name,
  children,
  index,
  setActiveIndex,
  active,
  topBorder,
  bottomBorder,
  last,
  arrowIcon,
}: AccordionRowProps) => {
  let iconActive: JSX.Element, iconInactive: JSX.Element;
  const handleSetIndex = () => {
    setActiveIndex(index);
  };

  if (arrowIcon) {
    iconActive = (
      <Icon
        name="icon-gui-disclosure-arrow"
        color="text-dark-grey"
        size="1.5rem"
        additionalCSS="-rotate-90"
      />
    );
    iconInactive = (
      <Icon
        name="icon-gui-disclosure-arrow"
        color="text-dark-grey"
        size="1.5rem"
        additionalCSS="rotate-90"
      />
    );
  } else {
    iconActive = (
      <Icon name="icon-gui-minus" color="text-dark-grey" size="1.5rem" />
    );
    iconInactive = (
      <Icon name="icon-gui-plus" color="text-dark-grey" size="1.5rem" />
    );
  }

  return (
    <div
      className={`border-mid-grey ${last && !bottomBorder ? "" : "border-b"} ${
        topBorder ? "border-t" : ""
      }`}
    >
      <button
        type="button"
        onClick={handleSetIndex}
        className={`flex w-full px-0 focus:outline-none py-20`}
      >
        <span className="ui-text-p1 font-bold text-left mr-8">{name}</span>
        <span className="ml-auto">{active ? iconActive : iconInactive}</span>
      </button>

      <section
        className="ui-text-p2 transition-all overflow-hidden"
        style={{
          maxHeight: active ? "500px" : "0",
          paddingBottom: active ? "1.5rem" : "0",
        }}
      >
        {children}
      </section>
    </div>
  );
};

const Accordion = ({
  data,
  id = "id-accordion",
  topBorder,
  bottomBorder,
  arrowIcon,
  autoClose,
  className,
}: AccordionProps) => {
  const [activeIndexes, setActiveIndexes] = useState<number[]>([]);

  const handleSetIndex = (index: number) => {
    const currentIndexIsActive = activeIndexes.includes(index);

    if (autoClose) {
      setActiveIndexes(currentIndexIsActive ? [] : [index]);
    } else {
      setActiveIndexes(
        currentIndexIsActive
          ? activeIndexes.filter((i) => i !== index)
          : [...activeIndexes, index],
      );
    }
  };

  return (
    <div className={className} id={id}>
      {data.map((item, currentIndex) => {
        return (
          <AccordionRow
            key={item.name}
            name={item.name}
            arrowIcon={arrowIcon}
            index={currentIndex}
            last={data.length === currentIndex + 1}
            topBorder={topBorder && currentIndex === 0}
            bottomBorder={bottomBorder && data.length === currentIndex + 1}
            active={activeIndexes.includes(currentIndex)}
            setActiveIndex={handleSetIndex}
          >
            {item.content}
          </AccordionRow>
        );
      })}
    </div>
  );
};

export default Accordion;
