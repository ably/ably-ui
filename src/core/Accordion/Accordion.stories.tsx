import React, { Fragment } from "react";
import Accordion, { AccordionProps } from "../Accordion";
import { IconName } from "../Icon/types";
import { accordionThemes } from "./types";

const loremText =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin scelerisque congue risus id lobortis. Vivamus blandit dolor at ultricies cursus. Phasellus pharetra nunc erat, quis porttitor mauris faucibus in. Donec feugiat dapibus orci et blandit. Duis eleifend accumsan est nec euismod. Proin imperdiet malesuada lacus, a aliquam eros aliquet nec. Sed eu dolor finibus, sodales nisl a, egestas mi. In semper interdum lacinia. Duis malesuada diam quis purus blandit, sit amet imperdiet neque accumsan. Morbi viverra vitae risus ut pellentesque. Praesent ac blandit augue. Aliquam purus lectus, lacinia in semper vitae, dictum eu felis. Donec vel pulvinar eros, id facilisis neque. Aenean odio arcu, accumsan vel est in, lobortis rhoncus ligula. Pellentesque sit amet odio velit.";

const lorem = <p className="mb-16">{loremText}</p>;

const textarea = (
  <textarea
    className="w-full h-256 bg-neutral-700 p-16 rounded-xl leading-relaxed"
    defaultValue={loremText}
  />
);

const data = [...Array(5)].map((_, i) => ({
  name: `Item ${i + 1}`,
  content: lorem,
}));

const dataWithIcons = data.map((datum) => ({
  ...datum,
  icon: "icon-gui-document-generic" as IconName,
}));

const dataWithTextarea = data.map((datum) => ({
  ...datum,
  content: textarea,
}));

const longData = data.map((datum) => ({
  ...datum,
  content: (
    <>
      {lorem}
      {lorem}
      {lorem}
      {lorem}
      {lorem}
    </>
  ),
}));

const AccordionPresentation = ({ data, options }: AccordionProps) => (
  <div className="grid sm:grid-cols-2 gap-16 w-full">
    {accordionThemes.map((theme) => (
      <div
        key={theme}
        className={`p-16 rounded-lg ${theme.includes("dark") ? "bg-neutral-1300" : ""}`}
      >
        <p className="ui-text-p3 mb-16 text-center font-mono">{theme}</p>
        <Accordion
          data={data}
          options={options}
          theme={theme}
          className="flex-1"
        />
      </div>
    ))}
  </div>
);

export default {
  title: "JS Components/Accordion",
  component: Accordion,
};

export const Default = { render: () => AccordionPresentation({ data }) };

export const AutoClose = {
  render: () =>
    AccordionPresentation({
      data,
      options: { autoClose: true },
    }),
  parameters: {
    docs: {
      description: {
        story:
          "Opening a section will close any other open sections. Set with `autoClose` on `options`.",
      },
    },
  },
};

export const SelectableHeaders = {
  render: () =>
    AccordionPresentation({
      data: dataWithIcons,
      options: { selectable: true },
    }),
  parameters: {
    docs: {
      description: {
        story:
          "Opening a section will apply styling to the opened section header. Row icons are disabled. Set with `selectable` on `options`.",
      },
    },
  },
};

export const StickyHeaders = {
  render: () => (
    <div className="h-[400px]">
      {AccordionPresentation({
        data: data,
        options: { sticky: true, defaultOpenIndexes: [0, 1, 2, 3] },
      })}
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story:
          "Section headers will fix to the top of the container when scrolled. Set with `sticky` on `options`.",
      },
    },
  },
};

export const WithDefaultOpenSections = {
  render: () =>
    AccordionPresentation({
      data: data,
      options: { defaultOpenIndexes: [1] },
    }),
  parameters: {
    docs: {
      description: {
        story:
          "The sections that correspond to the supplied indexes will be open by default. (e.g. `[1]` will open the second section). Set with `defaultOpenIndexes` on `options`.",
      },
    },
  },
};

export const WithRowIcons = {
  render: () =>
    AccordionPresentation({
      data: dataWithIcons,
    }),
  parameters: {
    docs: {
      description: {
        story:
          "When an icon name is supplied, it will be displayed to the left of the section name. Set with `icon` on `data` entries.",
      },
    },
  },
};

export const LongContent = {
  args: {
    data: longData,
  },
};

export const WithResizableInnerContent = {
  render: () =>
    AccordionPresentation({
      data: dataWithTextarea,
    }),
  parameters: {
    docs: {
      description: {
        story:
          "Try resizing content within the Accordion entries - the container should respond to the new height accordingly",
      },
    },
  },
};
