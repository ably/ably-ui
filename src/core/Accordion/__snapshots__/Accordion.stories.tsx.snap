// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Components/Accordion AutoClose smoke-test 1`] = `
<div id="id-accordion">
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 1
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 1
    </section>
  </div>
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 2
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 2
    </section>
  </div>
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 3
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 3
    </section>
  </div>
  <div class="border-mid-grey  ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 4
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 4
    </section>
  </div>
</div>
`;

exports[`JS Components/Accordion Default smoke-test 1`] = `
<div id="id-accordion">
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 1
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 1
    </section>
  </div>
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 2
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 2
    </section>
  </div>
  <div class="border-mid-grey border-b ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 3
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 3
    </section>
  </div>
  <div class="border-mid-grey  ">
    <button type="button"
            class="flex w-full px-0 focus:outline-none py-20"
    >
      <span class="ui-text-p1 font-bold text-left mr-8">
        Item 4
      </span>
      <span class="ml-auto">
        <svg class="text-dark-grey "
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-plus">
          </use>
        </svg>
      </span>
    </button>
    <section class="ui-text-p2 transition-all overflow-hidden"
             style="max-height: 0px; padding-bottom: 0px;"
    >
      Content 4
    </section>
  </div>
</div>
`;
